packages/debug-test/DebuggingFeatures.test.ts

```diff
@@ -0,0 +1,236 @@
+			request = http.request(options, response => response.on('data', data => resolve(data)));
+		});
+
+		request.end();
+
+		const data = await dataPromise;
+		expect(Buffer.isBuffer(data)).toBeTruthy();
+		const debugTargets = JSON.parse(data.toString());
+		expect(Array.isArray(debugTargets)).toBeTruthy();
+		expect(debugTargets.length).toBeGreaterThanOrEqual(1);
+		for (const debugTarget of debugTargets) {
+			expect(debugTarget.hasOwnProperty('id')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('title')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('faviconUrl')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('devtoolsFrontendUrl')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('type')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('webSocketDebuggerUrl')).toBeTruthy();
+			expect(debugTarget.hasOwnProperty('vm')).toBeTruthy();
+		}
+	}
+	finally {
+		await settings.uninitialize();
+	}
+});
+
+skip_test("debugger connection", async () => {
+
+	const settings = new PlaygroundDebugSettings({webDebugger: false, directDebugging: true, jsEngine: "Hermes"});
+	await settings.initialize();
+	try {
+		await selectPackage("Samples\\debugTest01");
+
+		const debugTargets = await getDebugTargets();
+		expect(debugTargets.length).toBe(2);
+
+		// testLog.message(`debugTarget0: URL: "${debugTargets[0].webSocketDebuggerUrl}", VM: "${debugTargets[0].vm}"`);
+		// testLog.message(`debugTarget1: URL: "${debugTargets[1].webSocketDebuggerUrl}", VM: "${debugTargets[1].vm}"`);
+		const ws = new WebSocket(debugTargets[0].webSocketDebuggerUrl);
+
+		ws.on('open', () => {
+			testLog.message(`WS on-open: sending Debugger.enable`);
+			ws.send(`{ "id":2, "method":"Debugger.enable", "params":{ "maxScriptsCacheSize":10000000 } }`)
+			testLog.message(`WS on-open: sending Debugger.pause`);
+			ws.send(`{ "id":11, "method":"Debugger.pause", "params":{}}`);
+		});
+
+
+		// 	{ "id":2, "method":"Debugger.enable", "params":{ "maxScriptsCacheSize":10000000 } }
+
+		ws.on('close', () => { testLog.message(`WS on-close`);});
+
+		ws.on('message', (data) => {
+			testLog.message(`WS on-message, typeof ${typeof data}`);
+
+			if(typeof data == 'string') {
+				testLog.message(`WS message data: "${data}"`);
+			}
+		});
+
+		await sleep(5000);
+
+		ws.close();
+
+	}
+	finally {
+		await settings.uninitialize();
+	}
+});
+
+test("enable-disable", async () => {
+	testLog.message(`test enable-disable`);
+
+	const settings = new PlaygroundDebugSettings({webDebugger: false, directDebugging: true, jsEngine: "Hermes"});
+	await settings.initialize();
+	try {
+		await selectPackage("Samples\\debugTest01");
+
+		const debugTargets = await getDebugTargets();
+		const dbg = new CDPDebugger(debugTargets[0].webSocketDebuggerUrl);
+
+		await dbg.DebuggerEnable();
+		await dbg.DebuggerDisable();
+		await dbg.checkOutstandingResponses(3000);
+
+		dbg.close();
+	}
+	finally {
+		await settings.uninitialize();
+	}
```

Comment:
It would be ideal to allow users of the API to avoid manual lifetime management.

One pattern I've used for this in other test code was modeled after `using` in C#, of lifetime controlled by a lexical scope.

```js
/**
 * Utility function to bind an instance of `PlaygroundDebugSettings` to a sync
 * or async function, uninitializing after function completion.
 * /
async function usingSettings(opts, fn) {
  const settings = new PlaygoundDebugSettings(opts);
  await settings.initialize();

  try {
    const ret = fn(settings);
    if (ret instanceof Promise) {
      await ret;
    }
  }
  finally {
    settings.uninitialized();
  }
}


// --- USAGE ---
await usingSettings(({webDebugger: false}, async (settings) => {
  // Test code using settings
});

```